packages/stores/src/account/osmosis/index.ts

```diff
@@ -2125,21 +2125,77 @@ export class OsmosisAccountImpl {
     );
   }
 
-  async sendAddAuthenticatorMsg(
-    authenticator: { type: string; data: any },
+  async sendAddOrRemoveAuthenticatorsMsg({
+    addAuthenticators,
+    removeAuthenticators,
+    memo = "",
+    onFulfill,
+  }: {
+    addAuthenticators: { type: string; data: Uint8Array }[];
+    removeAuthenticators: bigint[];
+    memo?: string;
+    onFulfill?: (tx: DeliverTxResponse) => void;
+  }) {
+    const addAuthenticatorMsgs = addAuthenticators.map((authenticator) =>
+      this.msgOpts.addAuthenticator.messageComposer({
+        type: authenticator.type,
+        data: authenticator.data,
+        sender: this.address,
+      })
+    );
+    const removeAuthenticatorMsgs = removeAuthenticators.map((id) =>
+      this.msgOpts.removeAuthenticator.messageComposer({
+        id,
+        sender: this.address,
+      })
+    );
+    const msgs = [...removeAuthenticatorMsgs, ...addAuthenticatorMsgs];
+
+    await this.base.signAndBroadcast(
+      this.chainId,
+      "addOrRemoveAuthenticators",
+      msgs,
+      memo,
+      undefined,
+      undefined,
+      (tx) => {
+        if (!tx.code) {
+          // Refresh the balances
+          const queries = this.queriesStore.get(this.chainId);
+
+          queries.queryBalances
+            .getQueryBech32Address(this.address)
+            .balances.forEach((balance) => balance.waitFreshResponse());
+
+          queries.cosmos.queryDelegations
+            .getQueryBech32Address(this.address)
+            .waitFreshResponse();
+
+          queries.cosmos.queryRewards
+            .getQueryBech32Address(this.address)
+            .waitFreshResponse();
+        }
```

Comment:
I'm assuming we'll eventually remove these for when we finish tRPC migration?

Reply:
That's right. In the future, I plan to eliminate these injected accounts and create message-composing functions. This way, we'll only need one function for broadcasting. This approach will simplify the process and offer the greatest flexibility for creating transactions, whether they have a single or multiple messages.